### ISSUE TITLE
PNI Category nav not accessible and invalid

### ISSUE THREAD

**Issue by tbrlpld** [MozillaFoundation/foundation.mozilla.org#11606]
### Describe the bug

I may be lacking the screen reader skills, but I was not able to properly interact with the "more categories" dropdown on PNI. 

See the screen recording to see me struggle to switch to any category that was hidden. E.g I was trying to activate "Cars" and "Toys & Games". Somehow I only kept getting "Reproductive Health". 

You can see that I get the other items to be read, but I cannot select them. Also, there is no delineation between the items. They feel just like one long string. 

I think the issue is, is that the whole menu is inside the toggle button. This feels very weird. Also, there is a list inside the button, which is invalid and was flagged by https://validator.w3.org/nu/?doc=https%3A%2F%2Ffoundation.mozilla.org%2Fen%2Fprivacynotincluded%2F (Related to #9875)

The validation issue is the nested `ul` inside of a `button`:
```
Error: Element ul not allowed as child of element button in this context. (Suppressing further errors from this subtree.)
From line 863, column 5; to line 863, column 168
/svg>↩    <ul id="pni-category-dropdown-select" class="tw-hidden tw-list-none tw-absolute tw-right-0 tw-p-0 tw-top-0 tw-z-50 tw-border-gray-20 tw-border tw-pb-4 tw-bg-white">↩     
Contexts in which element ul may be used:
Where flow content is expected.
Content model for element button:
Phrasing content, but there must be no interactive content descendant and no descendant with the tabindex attribute specified.
```

### To Reproduce
1. Go to https://foundation.mozilla.org/en/privacynotincluded/ (with Safari and VoiceOver)
2. Try to keyboard navigate (VO + arrow keys) into the "More categories"

### Expected behavior

After opening the menu, I should be able to navigate through them like a list. 

### Screenshots

https://github.com/MozillaFoundation/foundation.mozilla.org/assets/24797493/1a18d682-bb87-4636-8601-9f44d737fe9a

### Desktop (please complete the following information):
 - OS: macOS
 - Browser: Safari
 - Version: 14.2.1

**Comment by mmmavis** [comment]
tagging myself to QA on staging

**Comment by simont-cr** [comment]
Closing as the fix has already been merged to production. Thank you @mmmavis !
